projects/step-frontend/src/lib/modules/timeseries/performance-view/measurements/measurements-picker.component.html

```diff
@@ -1,42 +1,31 @@
-<button
-  mat-icon-button
-  [matTooltip]="'Keywords (' + activeKeywords + ' active)'"
-  [matMenuTriggerFor]="menu"
-  style="min-width: 20px"
->
-  <!--  Measurements {{ activeKeywords ? '(' + activeKeywords + ')' : '' }}-->
-  <step-icon name="settings"></step-icon>
-</button>
-<mat-menu #menu="matMenu" class="time-selector-menu">
-  <div (click)="$event.stopPropagation()" class="menu-container">
-    <div>
-      <input [(ngModel)]="keywordSearchValue" placeholder="Search" style="width: 110px; margin-bottom: 12px" />
-    </div>
-    <div>
-      <input
-        type="checkbox"
-        class="checkbox"
-        [(ngModel)]="allSeriesChecked"
-        (change)="onAllSeriesCheckboxClick($event)"
-        style="display: inline"
-      />
-      <span>All keywords</span>
-    </div>
-    <ng-container *ngFor="let item of keywords | keyvalue: valueAscOrder">
-      <ng-container *ngIf="!keywordSearchValue || item.key.toLowerCase().indexOf(keywordSearchValue) >= 0">
-        <div class="color-cell" *ngIf="item.value.isVisible">
-          <input
-            type="checkbox"
-            [ngModel]="item.value.isSelected"
-            class="checkbox"
-            (change)="onKeywordToggle(item.key, $event)"
-          />
-          <div style="display: flex; align-items: center; margin-top: 3px">
-            <span class="color-square" [style.background-color]="item.value.color"></span>
-            <span>{{ item.key }}</span>
-          </div>
+<div (click)="$event.stopPropagation()" class="menu-container">
+  <div>
+    <input [(ngModel)]="keywordSearchValue" placeholder="Search" style="width: 110px; margin-bottom: 12px" />
+  </div>
+  <div>
+    <input
+      type="checkbox"
+      class="checkbox"
+      [(ngModel)]="allSeriesChecked"
+      (change)="onAllSeriesCheckboxClick($event)"
+      style="display: inline"
+    />
+    <span>All keywords</span>
+  </div>
+  <ng-container *ngFor="let item of keywords | keyvalue: valueAscOrder">
+    <ng-container *ngIf="!keywordSearchValue || item.key.toLowerCase().indexOf(keywordSearchValue) >= 0">
```

Comment:
It's not optimal.
Iteration has done over all entities, and condition is checked for every item during each change detection cycle.
This can be solved in the following way.

In the component create a form control
```
readonly keywordSearchValueCtrl: FormControl;

constructor(
  formBuilder: FormBuilder
){
  this.keywordSearchValueCtrl = formBuilder.control('');
}

```
Then bind it to input
```
<input [formControl]="keywordSearchValueCtrl" placeholder="Search" style="width: 110px; margin-bottom: 12px" />
```

In component create a rxjs stream of filtered items, that is based on controls'value:
```
readonly keywordItems$ = this.keywordSearchValueCtrl.valueChanges.pipe(
  startWith(''),
  distinctUntilChanged(),
  map((searchValue) => {
      // make a filtration here
  }) 
)
```
In template iterate over items in stream
```
<ng-container *ngFor="let item of (keywordItems$ | async)">
...
</ng-container>
```

Reply:
Solved! keyvalue pipe is a problematic one since it's NOT pure. I created a new pipe which transform and filter